akka-stream/src/main/scala/akka/stream/impl/PhasedFusingActorMaterializer.scala

```diff
@@ -613,6 +613,9 @@ final class GraphStageIsland(
 
   def conn(slot: Int): Connection = {
     maxConnections = math.max(slot, maxConnections)
+    if (maxConnections >= connections.length) {
+      connections = connections.padTo(connections.length * 2, null)
```

Comment:
might not be done very often, but this is more efficient:

    connections = java.util.Arrays.copyOf(connections, connections.length * 2)